framework/src/transfers/MultiAppFieldTransfer.C

```diff
@@ -0,0 +1,443 @@
+                                   _from_postprocessors_to_be_preserved[0],
+                                   to_problem,
+                                   _to_postprocessors_to_be_preserved[i]);
+      }
+
+      // Compute the to-postproessor again so that it has the right value with the updated solution
+      if (_use_nearestpoint_pps)
+        to_problem.computeUserObjectByName(EXEC_TRANSFER, _to_postprocessors_to_be_preserved[0]);
+    }
+
+    _console << "Finished Conservative transfers " << name() << std::endl;
+  }
+}
+
+void
+MultiAppFieldTransfer::adjustTransferedSolutionNearestPoint(unsigned int i,
+                                                            FEProblemBase * from_problem,
+                                                            PostprocessorName & from_postprocessor,
+                                                            FEProblemBase & to_problem,
+                                                            PostprocessorName & to_postprocessor)
+{
+  PostprocessorValue from_adjuster = 0;
+  if (from_problem && _direction == FROM_MULTIAPP)
+    from_adjuster = from_problem->getPostprocessorValue(from_postprocessor);
+  else
+    from_adjuster = 0;
+
+  /* Everyone on master side should know this value, and use it to scale the solution */
+  if (_direction == FROM_MULTIAPP)
+  {
+    if (from_adjuster < 0.0)
+      mooseError("from_adjuster cannot be negative ");
+
+    /* This might not be right if the integral is negative */
+    comm().max(from_adjuster);
+  }
+
+  PostprocessorValue to_adjuster = 0;
+  // Compute to-postproessor to have the adjuster
+  if (_direction == TO_MULTIAPP)
+  {
+    to_problem.computeUserObjectByName(EXEC_TRANSFER, to_postprocessor);
+    to_adjuster = to_problem.getPostprocessorValue(to_postprocessor);
+  }
+
+  auto & to_var = to_problem.getVariable(
+      0, _to_var_name, Moose::VarKindType::VAR_ANY, Moose::VarFieldType::VAR_FIELD_STANDARD);
+  auto & to_sys = to_var.sys().system();
+  auto var_num = to_sys.variable_number(_to_var_name);
+  auto sys_num = to_sys.number();
+  auto & pps = static_cast<const NearestPointIntegralVariablePostprocessor &>(
+      _direction == FROM_MULTIAPP ? (to_problem.getUserObjectBase(to_postprocessor))
+                                  : (from_problem->getUserObjectBase(from_postprocessor)));
+  auto & to_solution = to_var.sys().solution();
+  auto & to_mesh = to_problem.mesh().getMesh();
+  bool is_nodal = to_sys.variable_type(var_num).family == LAGRANGE;
+  if (is_nodal)
+  {
+    for (const auto & node : to_mesh.local_node_ptr_range())
+    {
+      // Skip this node if the variable has no dofs at it.
+      if (node->n_dofs(sys_num, var_num) < 1)
+        continue;
+
+      Real scale = 1;
+      if (_direction == FROM_MULTIAPP)
+      {
+        auto ii = pps.nearestPointIndex(*node);
+        if (ii != i)
+          continue;
+        if (pps.userObjectValue(i) == 0.0)
```

Comment:
Equality checks with Real values aren't the best. Could we do a really small abs check with a tolerance. Or perhaps use one of the MOOSE "fuzzy" checks in the utils?